functions/Test-DbaSqlBuild.ps1

```diff
@@ -254,19 +254,21 @@ function Test-DbaSqlBuild {
                             $targetSP = 0
                         }
                         $targetSPName = $AllSPs[$targetSP]
-                        Write-Message -Level verbose -Message "Target SP is $targetSPName - $targetSP on $($AllSPs.Length)"
+                        Write-Message -Level Debug -Message "Target SP is $targetSPName - $targetSP on $($AllSPs.Length)"
                         $targetedBuild = $SPsAndCUs | Where-Object SP -eq $targetSPName | Select-Object -First 1
                     }
                     if ($ParsedMaxBehind.ContainsKey('CU')) {
-                        $SPsAndCUs | Where-Object Version -gt $targetedBuild.Version | ConvertTo-Json
                         $AllCUs = ($SPsAndCUs | Where-Object VersionObject -gt $targetedBuild.VersionObject).CU | Select-Object -Unique
-                        $targetCU = $AllCUs.Length - $ParsedMaxBehind['CU'] - 1
-                        if ($targetCU -lt 0) {
-                            $targetCU = 0
+                        if ($AllCUs.Length -gt 0) {
+                            #CU after the targeted build available
+                            $targetCU = $AllCUs.Length - $ParsedMaxBehind['CU'] - 1
+                            if ($targetCU -lt 0) {
+                                $targetCU = 0
+                            }
+                            $targetCUName = $AllCUs[$targetCU]
+                            Write-Message -Level Debug -Message "Target CU is $targetCUName - $targetCU on $($AllCUs.Length)"
```

Comment:
Needs to be `$($AllCUs.Length -1)`

Reply:
same as previous

Reply:
Forget this comment. Here this is right!